libcasr/src/stacktrace.rs

```diff
@@ -288,26 +288,35 @@ impl Filter for Stacktrace {
         }
 
         // Find repeating intervals in stacktrace
-        let mut indices = Vec::new();
-        indices.resize(self.len(), true);
         // We need to create an element that will not be equal to others
         let mut not_an_element = StacktraceEntry::default();
         not_an_element.debug.file = "CASR_not_equal_to_other_names_CASR".to_string();
         not_an_element.debug.line = 1;
-        if let Ok(intervals) = get_interval_repetitions(self, &not_an_element) {
-            intervals.iter().for_each(|(start, end, seq_len)| {
-                (start + seq_len..end + 1).for_each(|idx| indices[idx] = false)
-            });
-        }
-        let mut keep = indices.iter();
+        let mut intervals = get_interval_repetitions(self, &not_an_element).unwrap_or_default();
+        intervals.sort_by(|x, y| x.0.partial_cmp(&y.0).unwrap());
```

Comment:
why not regular cmp instead of partial_cmp?

Reply:
changed